x/market/client/cli/parse.go

```diff
@@ -30,3 +49,18 @@ func (input *createDealInputs) UnmarshalJSON(data []byte) error {
 	*input = createDealInputs(createDeal)
 	return nil
 }
+
+type XSellDataInputs sellDataInputs
+
+func (input *sellDataInputs) UnmarshalJSON(data []byte) error {
+	var sellData XSellDataInputs
+	dec := json.NewDecoder(bytes.NewReader(data))
+	dec.DisallowUnknownFields()
+
+	if err := dec.Decode(&sellData); err != nil {
+		return err
+	}
+
+	*input = sellDataInputs(sellData)
+	return nil
+}
```

Comment:
```suggestion
func (input *sellDataInputs) UnmarshalJSON(data []byte) error {
	var sellData sellDataInputs
	dec := json.NewDecoder(bytes.NewReader(data))
	dec.DisallowUnknownFields()

	if err := dec.Decode(&sellData); err != nil {
		return err
	}

	*input = sellData
	return nil
}
```
I think we don't need `X...` structs. Please check if my code works. If so, we can also delete the `XCreateDealInputs` above.

Reply:
Agreed